server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/ActiveRuleDao.java

```diff
@@ -101,6 +103,16 @@ public void delete(DbSession session, ActiveRuleKey key) {
     }
   }
 
+  public void deleteByProfileKeys(DbSession dbSession, Collection<String> profileKeys) {
+    ActiveRuleMapper mapper = mapper(dbSession);
+    DatabaseUtils.executeLargeUpdates(profileKeys, mapper::deleteByProfileKeys);
+  }
+
+  public void deleteParametersByProfileKeys(DbSession dbSession, Collection<String> profileKeys) {
+    ActiveRuleMapper mapper = mapper(dbSession);
+    DatabaseUtils.executeLargeUpdates(profileKeys, mapper::deleteParametersByProfileKeys);
+  }
+
```

Comment:
There are no rules, only guidelines ! It's for sure possible to merge, but splitting them allows to split unit tests as well.